docs/configuring-playbook-matrix-media-repo.md

```diff
@@ -1,17 +1,25 @@
 # Setting up matrix-media-repo (optional)
 
-[matrix-media-repo](https://docs.t2bot.io/matrix-media-repo/) is a highly customizable multi-domain media repository for Matrix. Intended for medium to large environments consisting of several homeservers, this media repo de-duplicates media (including remote media) while being fully compliant with the specification.
+[matrix-media-repo](https://docs.t2bot.io/matrix-media-repo/) (often abbreviated "MMR") is a highly customizable multi-domain media repository for Matrix. Intended for medium to large environments consisting of several homeservers, this media repo de-duplicates media (including remote media) while being fully compliant with the specification.
 
 Smaller/individual homeservers can still make use of this project's features, though it may be difficult to set up or have higher than expected resource consumption. Please do your research before deploying this as this project may not be useful for your environment.
 
 For a simpler alternative (which allows you to offload your media repository storage to S3, etc.), you can [configure S3 storage](configuring-playbook-s3.md) instead of setting up matrix-media-repo.
 
+| **Table of Contents**                                                                       |
+| :------------------------------------------------------------------------------------------ |
+| [Quickstart](#quickstart)                                                                   |
+| [Additional configuration options](#configuring-the-media-repo)                             |
+| [Importing data from an existing media store](#importing-data-from-an-existing-media-store) |
+
 ## Quickstart
 
 Add the following configuration to your `inventory/host_vars/matrix.DOMAIN/vars.yml` file:
 
 ```yaml
 matrix_media_repo_enabled: true
+# Do NOT put your homeserver's existing database credentials here, create a NEW password:
+matrix_media_repo_database_password: "your_password"
```

Comment:
Database passwords for all components are auto-generated (derived from `matrix_homeserver_generic_secret_key`), so that people do not need to define them manually.

You can see this done for all of them, including `matrix_media_repo_database_password` in `group_vars/matrix_servers`. Example:

https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/8b9faa7e6575af176e6f1e00a3b309c52039f46f/group_vars/matrix_servers#L2694

I understand you'd like to hardcode your own password here, so you can use it in your "importing flow" below, but I think we'd better address it in another way. Users can find their MMR password in its configuration. We could provide a `grep ..` command that shows it to them. We may also extend the playbook with some `dump-variable` tag which would expose a variable.

Reply:
Ah, my bad. I only saw that there was an insecure default provided in the `roles/custom/matrix-media-repo/defaults/main.yml` and didn't realize it was replaced properly elsewhere.

Also, I did a little more testing and I think I realized a much cleaner way to do this anyways. One can just use the `matrix` superuser and the `devture_postgres_connection_password` that is already manually set in vars.yml, which conveniently also avoids any need to adjust database permissions.

I'll update with all these fixes shortly.

Reply:
The downside of this is that you're trusting matrix-media-repo with your database superuser credentials, but.. it's probably fine. At least it's a one-off operation and MMR is not using these credentials at all times later. Still, it's putting a lot of trust in it not to do damage.

I suppose it's a good solution that is available to everyone - one that doesn't involve modifying the database grants.